.github/workflows/check-manifest.yaml

```diff
@@ -0,0 +1,26 @@
+name: check MANIFEST.in
+on: [pull_request]
```

Comment:
Do we want this on every PR?

Reply:
For lack of a smarter approach, I think so ? Would you prefer we ran this at a fixed frequency?

Reply:
I think I'm maybe reacting to seeing all the existing github workflows and how they can, at times, be a bit intimidating.

Reply:
FWIW the h5py one has seen so little failures I'm aiming at getting rid of it soon :)